src/Message/PurchaseResponse.php

```diff
@@ -1,113 +1,108 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Omnipay\SchoolEasyPay\Message;
 
 use Omnipay\Common\Message\AbstractResponse;
+use Symfony\Component\HttpFoundation\Response;
 
 /**
  * Response class for all SchoolEasyPay requests
  */
 class PurchaseResponse extends AbstractResponse
 {
-    /** @var string Request ID */
-    protected $requestId = null;
-    /** @var string HTTP response code */
-    protected $httpResponseCode = null;
+    protected string $requestId;
+
+    protected string $httpResponseCode;
 
     /**
      * Is the transaction successful?
-     * @return boolean True if successful
      */
-    public function isSuccessful()
+    public function isSuccessful(): bool
     {
         // get response code
         $code = $this->getHttpResponseCode();
 
-        if ($code === 200) {  // OK
-            return true;
-        }
-
-        if ($code === 201 && $this->isApproved()) {   // Created
-            return true;
-        }
-
-        if ($code === 202 && $this->isPending()) {   // Accepted
+        if ($code === '200' || ($code === '201' && $this->isApproved())) {
             return true;
         }
 
-        return false;
+        // Accepted
+        return $code === '202' && $this->isPending();
     }
 
     /**
      * Is the transaction approved?
-     * @return boolean True if approved
      */
-    public function isApproved()
+    public function isApproved(): bool
     {
         return $this->getStatus() === 'Successful';
     }
 
     /**
      * Is the transaction pending?
-     * @return boolean True if pending
      */
-    public function isPending()
+    public function isPending(): bool
     {
         return $this->getStatus() === 'Pending';
     }
 
     /**
      * Get Transaction reference
-     * @return string Payway transaction reference
      */
-    public function getTransactionReference()
+    public function getTransactionReference(): string
     {
-        return $this->getData('paymentReference');
+        return $this->getDataItem('paymentReference');
     }
 
-    public function getSettlementDate()
+    public function getSettlementDate(): string
     {
-        return $this->getData('settlementDate');
+        return $this->getDataItem('settlementDate');
     }
 
     /**
      * Get status
-     * @return array Returned status
      */
-    public function getStatus()
+    public function getStatus(): string
     {
-        return $this->getData('paymentStatus');
+        return $this->getDataItem('paymentStatus');
+    }
+
+    public function getDataItem(string $key): ?string
+    {
+        $data = $this->getData();
+
+        return $data[$key] ?? null;
     }
 
     /**
-     * Get response data, optionally by key
-     * @param  string       $key Data array key
-     * @return string|array      Response data item or all data if no key specified
+     * Get all response data
      */
-    public function getData($key = null)
+    public function getData(): array
```

Comment:
Would this always be an array?
https://github.com/thephpleague/omnipay-common/blob/master/src/Common/Message/AbstractResponse.php#L45

Reply:
It was previously treated as an array
```
    /**
     * Get response data, optionally by key
     * @param  string       $key Data array key
     * @return string|array      Response data item or all data if no key specified
     */
    public function getData($key = null)
    {
        if ($key) {
            return isset($this->data[$key]) ? $this->data[$key] : null;
        }
        return $this->data;
    }
```
I've broken this out into two methods because it was previously serving two purposes (ie get all data or get a particular data item).

The omnipay/common package hasn't been updated in 3 years so plenty of it is still untyped, so I've needed to improvise a bit to make our packages properly type hinted.

Reply:
Just wondering if there is any way for that to not be an array since the omnipay package says its `mixed`.

Reply:
I kind of see what you mean.  I don't think there's anything inherently wrong with requiring `data` to be an array in this class since we're just narrowing the scope of accepted type from the parent, however, I understand that we may get some odd behaviour if we provide a non-array on instantiation since the parent constructor accepts any type.  So, to clear that up I have added a constructor to this class that throws an exception if a non-array is passed to it and then calls the parent constructor.  This allows the property to still be defined as `mixed` but to have the array type enforced in this implementation.